cmds/tui.c

```diff
@@ -295,12 +301,78 @@ static struct display_field *graph_field_table[] = {
 	&graph_field_addr,
 };
 
+#define REPORT_FIELD_STRUCT(_id, _name, _func, _header, _length)	\
+static struct display_field report_field_##_func = {			\
+	.id      = _id,							\
+	.name    = #_name,						\
+	.header  = _header,						\
+	.length  = _length,						\
+	.print   = print_report_##_func,				\
+	.list    = LIST_HEAD_INIT(report_field_##_func.list)		\
+};
+
+#define REPORT_FIELD_TIME(_id, _name, _field, _func, _header)	\
+static void print_report_##_func(struct field_data *fd)		\
+{									\
+	struct uftrace_report_node *node = fd->arg;			\
+	uint64_t d = node->_field;					\
+	printw("  ");							\
+	print_time(d);							\
+}									\
+REPORT_FIELD_STRUCT(_id, _name, _func, _header, 11)
+
+#define REPORT_FIELD_CALL(_id, _name, _field, _func, _header)	\
+static void print_report_##_func(struct field_data *fd)		\
+{									\
+	struct uftrace_report_node *node = fd->arg;			\
+	uint64_t d = node->_field;					\
+	printw("  ");							\
+	printw("%10"PRIu64 "", d);					\
+}									\
+REPORT_FIELD_STRUCT(_id, _name, _func, _header, 11)
+
+REPORT_FIELD_TIME(REPORT_F_TOTAL_TIME, total, total.sum, total, "TOTAL TIME");
+REPORT_FIELD_TIME(REPORT_F_TOTAL_TIME_AVG, total-avg, total.avg, total_avg, "TOTAL AVG");
+REPORT_FIELD_TIME(REPORT_F_TOTAL_TIME_MIN, total-min, total.min, total_min, "TOTAL MIN");
+REPORT_FIELD_TIME(REPORT_F_TOTAL_TIME_MAX, total-max, total.max, total_max, "TOTAL MAX");
+REPORT_FIELD_TIME(REPORT_F_SELF_TIME, self, self.sum, self, "SELF TIME");
+REPORT_FIELD_TIME(REPORT_F_SELF_TIME_AVG, self-avg, self.avg, self_avg, "SELF AVG");
+REPORT_FIELD_TIME(REPORT_F_SELF_TIME_MIN, self-min, self.min, self_min, "SELF MIN");
+REPORT_FIELD_TIME(REPORT_F_SELF_TIME_MAX, self-max, self.max, self_max, "SELF MAX");
+REPORT_FIELD_CALL(REPORT_F_CALL, call, call, call, "CALL");
+
+static struct display_field *report_field_table[] = {
+	&report_field_total,
+	&report_field_total_avg,
+	&report_field_total_min,
+	&report_field_total_max,
+	&report_field_self,
+	&report_field_self_avg,
+	&report_field_self_min,
+	&report_field_self_max,
+	&report_field_call,
+};
+
 static void setup_default_graph_field(struct list_head *fields, struct opts *opts,
 				      struct display_field *p_field_table[])
 {
 	add_field(fields, p_field_table[GRAPH_F_TOTAL_TIME]);
 }
 
+static void setup_default_report_field(struct list_head *fields, struct opts *opts,
+				      struct display_field *p_field_table[])
+{
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME]);
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_AVG]);
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_MIN]);
+	add_field(fields, p_field_table[REPORT_F_TOTAL_TIME_MAX]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_AVG]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_MIN]);
+	add_field(fields, p_field_table[REPORT_F_SELF_TIME_MAX]);
+	add_field(fields, p_field_table[REPORT_F_CALL]);
```

Comment:
You added all fields to the output list which changes the behavior.  That's not what I want.  I'd like to see the fields that are enabled as of now.  So let's create all fields (like you did) but add 'total', 'self' and 'call' only.